ide/svg/src/org/netbeans/modules/svg/SVGViewerElement.java

```diff
@@ -0,0 +1,361 @@
+    }
+
+    @Override
+    public JComponent getVisualRepresentation() {
+        if (component == null) {
+            viewer = new JPanel();
+            component = viewer;
+        }
+
+        return component;
+    }
+
+    @Override
+    public JComponent getToolbarRepresentation() {
+        if (toolbar == null) {
+            toolbar = new SVGViewerToolbar().createToolbar(this);
+        }
+
+        return toolbar;
+    }
+
+    @Override
+    public Action[] getActions() {
+        return callback.createDefaultActions();
+    }
+
+    @Override
+    public Lookup getLookup() {
+        return lookup;
+    }
+
+    @Override
+    public void componentOpened() {
+        dataObject.getPrimaryFile().addFileChangeListener(fcl);
+
+        updateView();
+    }
+
+    @Override
+    public void componentClosed() {
+        dataObject.getPrimaryFile().removeFileChangeListener(fcl);
+    }
+
+    @Override
+    public void componentShowing() {
+    }
+
+    @Override
+    public void componentHidden() {
+    }
+
+    @Override
+    public void componentActivated() {
+    }
+
+    @Override
+    public void componentDeactivated() {
+    }
+
+    @Override
+    public UndoRedo getUndoRedo() {
+        return UndoRedo.NONE;
+    }
+
+    @Override
+    public void setMultiViewCallback(MultiViewElementCallback callback) {
+        this.callback = callback;
+    }
+
+    @Override
+    public CloseOperationState canCloseElement() {
+        return CloseOperationState.STATE_OK;
+    }
+
+    private void addMouseWheelListenerToViewer(JScrollPane scrollPane) {
+        svgPanel.addMouseWheelListener(e -> {
+            double oldScale = scale;
+
+            // Point in scrolled pane
+            Point visiblePoint = e.getPoint();
+
+            // "Picturepixel"
+            Point markedPoint = new Point(
+                (int) (visiblePoint.getX() / oldScale),
+                (int) (visiblePoint.getY() / oldScale));
+
+            int clicks = e.getWheelRotation();
+            int clicks_abs = Math.abs(clicks);
+            for (int i = 0; i < clicks_abs; i++) {
+                if (clicks < 0) {
+                    zoomIn();
+                } else {
+                    zoomOut();
+                }
+            }
+
+            double newScale = scale;
+
+            Point markedPointInRealSpace = new Point(
+                (int) (markedPoint.getX() * newScale),
+                (int) (markedPoint.getY() * newScale)
+            );
+
+            Rectangle r = scrollPane.getViewport().getViewRect();
+
+            r.setLocation(markedPointInRealSpace);
+            r.translate(-r.width / 2, -r.height / 2);
+
+            svgPanel.scrollRectToVisible(r);
+        });
+    }
+
+    private void updateView() {
+        FileObject fo = dataObject.getPrimaryFile();
+
+        if (fo == null) {
+            return;
+        }
+
+        if (viewer == null) {
+            return;
+        }
+
+        svgDocument = svgLoader.load(fo.toURL());
```

Comment:
@weisJ will this load file-relative resources by default? if yes we should turn this off by default IMO.

was also mentioned in https://github.com/apache/netbeans/pull/7938#issuecomment-2466319810, cc @eirikbakke

Reply:
No, loading of external resources needs to be enabled explicitly.

Reply:
@weisJ Great, that's the desired behavior!
@mbien There's some extra logic in SVGIcon.java to deal with externally loaded resources, but it's only to force an exception to be thrown in these cases in order to make some existing unit tests pass. That's not needed in this PR.